src/platform/packages/private/kbn-grid-layout/grid/grid_section/delete_grid_section_modal.tsx

```diff
@@ -56,41 +58,51 @@ export const DeleteGridRowModal = ({
             setDeleteModalVisible(false);
           }}
         >
-          {i18n.translate('kbnGridLayout.deleteGridRowModal.cancelButton', {
+          {i18n.translate('kbnGridLayout.deleteGridSectionModal.cancelButton', {
             defaultMessage: 'Cancel',
           })}
         </EuiButtonEmpty>
         <EuiButton
           onClick={() => {
             setDeleteModalVisible(false);
             const layout = gridLayoutStateManager.gridLayout$.getValue();
-            const firstRowId = Object.values(layout).find(({ order }) => order === 0)?.id;
-            if (!firstRowId) return;
-            let newLayout = movePanelsToRow(layout, rowId, firstRowId);
-            newLayout = deleteRow(newLayout, rowId);
-            gridLayoutStateManager.gridLayout$.next(newLayout);
+            if (layout[sectionId].isMainSection) return; // main sections are not user deletable
+
+            // convert collapsible section to main section so that panels remain in place
+            const newLayout = cloneDeep(layout);
+            const { title, isCollapsed, ...sectionAsMain } = {
+              ...(newLayout[sectionId] as CollapsibleSection),
+              isMainSection: true,
+            };
+            newLayout[sectionId] = sectionAsMain as MainSection;
+
+            gridLayoutStateManager.gridLayout$.next(resolveSections(newLayout));
```

Comment:
to avoid casting, we could do:

```suggestion
              const isCollapsibleSection = (section: CollapsibleSection | MainSection): section is CollapsibleSection => !section.isMainSection; // this probably should be moved to top of this file or another helpers file

            const section = layout[sectionId];


            if (!isCollapsibleSection(section)) return; // main sections are not user deletable

            // convert collapsible section to main section so that panels remain in place
            const newLayout = cloneDeep(layout);
            const { title, isCollapsed, ...baseSectionProps } = section
            const sectionAsMain: MainSection = {
              ...baseSectionProps,
              isMainSection: true,
            }

            newLayout[sectionId] = sectionAsMain; 

            gridLayoutStateManager.gridLayout$.next(resolveSections(newLayout));
```